### PR TITLE
[Topi & Relay] Add quantization support for the vision transform model in GPU

### PR DESCRIPTION
**Pull request by huochaitiantang**
We submit this PR to add quantization support for the [vision transform (vit)](https://arxiv.org/abs/2010.11929) model in GPU. The main change is as follows:

1, In vit model,  time-consuming operators are batch_matmul,  so we first add the compute and schedule of `batch_matmul_int8.cuda` in tvm.topi.cuda

2, To support the quantization of batch_matmul, we then add `batch_matmul_rewrite` and `BatchMatmulRealize` in tvm.relay.quantize 

3, The kl -divergence calibrate could not preserve the accuracy of vit model well, so we add the `_percentile_scale` function 

For the vit-B32-224 model, the performance is as follows:

- Top-1 accuracy in Imagenet validation
  - paper: 73.38
  - nonofficial-model-fp32: 73.27
  - nonofficial-model-int8: 72.78

- The latency in GTX1660 GPU
  - nonofficial-model-fp32: 10.32 ms
  - nonofficial-model-int8: 4.93 ms

Thanks for your review! @jcf94 @tqchen

### CODE REVIEW

**Comment by XHPlus** [comment]
Thanks for the reviewer. We will keep updating more results for other ViT models and contributing more quantization calibration algorithms.

**Review by jcf94 — CHANGES_REQUESTED** [review]
Thanks! @huochaitiantang
This PR looks great! I have only few comments.

File: python/tvm/relay/op/strategy/cuda.py [hunk 1]
```diff
@@ -722,12 +722,20 @@ def dense_strategy_cuda(attrs, inputs, out_type, target):
...
-        wrap_topi_schedule(topi.cuda.schedule_batch_matmul),
-        name="batch_matmul.cuda",
-        plevel=10,
-    )
+    x, y = inputs
+    if x.dtype == "int8" and y.dtype == "int8" and out_type.dtype == "int32":
+        strategy.add_implementation(
+            wrap_compute_batch_matmul(topi.cuda.batch_matmul_int8, need_out_dtype=True),
+            wrap_topi_schedule(topi.cuda.schedule_batch_matmul_int8),
+            name="batch_matmul_int8.cuda",
```
[the remark this answers is not in the document]
Comment (huochaitiantang) [reply python/tvm/relay/op/strategy/cuda.py:730]: We add plevel=10 for the batch_matmul_int8.cuda, which is the same as dense_int8.cuda

File: tests/python/nightly/quantization/test_quantization_accuracy_for_vit.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+def get_onnx_model(model_name, batch_size, qconfig, target=None, original=False, dataset=None):
+
+    assert model_name == "vit32", "Only support vit32 model!"
+    logfile = "gtx1660_vit_B32_224.log"
+    onnx_path = "vit_B32_224.onnx"
+
+    if not os.path.exists(logfile):
+        os.system("wget https://github.com/TheGreatCold/tvm-vit/raw/master/{}".format(logfile))
+    if not os.path.exists(onnx_path):
+        os.system("wget https://github.com/TheGreatCold/tvm-vit/raw/master/{}".format(onnx_path))
```
[the remark this answers is not in the document]
Comment (huochaitiantang) [reply tests/python/nightly/quantization/test_quantization_accuracy_for_vit.py]: Thanks for your review. We updated the download codes based on your suggestion. Besides, the wget method is not compatible on different platforms, so we use the urllib library instead.

**Review by jcf94 — APPROVED** [review]
Thanks! @huochaitiantang @XHPlus . Have fun with TVM. Looking forward to see more contributions from you!

**Pull request merged**